src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -69,21 +69,41 @@ public long getRowCountForScale(int scale)
         }
     }
 
-    private long computeCountUsingStaticScale()
+    private long computeCountUsingStaticScale(float scale)
     {
-        return getRowCountForScale(1);
+        if (scale == Math.round(scale)) {
+            return getRowCountForScale(1);
+        }
+        // Handle decimal scales (not part of spec). Will only have an effect on
+        // decimal scales 0 < x < 1.
```

Comment:
For static scales, the row counts for all scales other than 0 are the same. That means that when you're scaling at say 3.4, the row counts at scale 1 and scale 10 (the two nearest defined scales) will be exactly the same, making `delta` equal to zero, reducing the computation to what it usually does. However, for a scale 0 < x < 1 the two nearest defined scales will be different making `delta` non-zero.

I now realize the above if statement is a little redundant. I guess I left it there to make it obvious to someone else reading the code that for integer scales we preserve spec behavior. Otherwise they would have to go through the reasoning I posted above to realize that for scales not in 0 < x < 1 the code is reduced to the previous behavior.